### PR TITLE
[twitter] fixed duration

### PR DESCRIPTION
**Pull request by capital-G**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

The duration of Twitter Videos was not in seconds, for example this Video:
https://twitter.com/SophiePassmann/status/903676793860829184

Old
```
python -m youtube_dl --get-duration https://twitter.com/SophiePassmann/status/903676793860829184
29:50:00
```
After fix:
```
python -m youtube_dl --get-duration https://twitter.com/SophiePassmann/status/903676793860829184
1:47
```

### CODE REVIEW

File: youtube_dl/extractor/twitter.py [hunk 1]
```diff
@@ -229,7 +229,7 @@ def _real_extract(self, url):
 
         title = self._search_regex(r'<title>([^<]+)</title>', webpage, 'title')
         thumbnail = config.get('posterImageUrl') or config.get('image_src')
-        duration = float_or_none(config.get('duration')) or duration
+        duration = float_or_none(config.get('duration') * 10e-4) or duration
```
Comment (dstftw) [inline youtube_dl/extractor/twitter.py]: Breaks extraction if `config.get('duration')` is not numeric.

**Comment by capital-G** [comment]
Should be fixed now

File: youtube_dl/extractor/twitter.py [hunk 2]
```diff
@@ -229,7 +229,10 @@ def _real_extract(self, url):
 
         title = self._search_regex(r'<title>([^<]+)</title>', webpage, 'title')
         thumbnail = config.get('posterImageUrl') or config.get('image_src')
-        duration = float_or_none(config.get('duration')) or duration
+        if config.get('duration'):
+            duration = float_or_none(config.get(duration))
+            if duration:
+                duration *= 10e-4
```
Comment (dstftw) [inline youtube_dl/extractor/twitter.py]: 1. Do not remove fallback to previous duration value.
2. `scale` of `float_or_none` instead.

**Comment by capital-G** [comment]
I used the inbuilt scale and the fallback for duration now.

File: youtube_dl/extractor/twitter.py [hunk 3]
```diff
@@ -229,7 +229,8 @@ def _real_extract(self, url):
 
         title = self._search_regex(r'<title>([^<]+)</title>', webpage, 'title')
         thumbnail = config.get('posterImageUrl') or config.get('image_src')
-        duration = float_or_none(config.get('duration')) or duration
+        if config.get('duration'):
```
Comment (dstftw) [inline youtube_dl/extractor/twitter.py]: This is already checked in `float_or_none`.

**Comment by capital-G** [comment]
You're right, fixed it

**Comment by dstftw** [comment]
Add duration to tests.

**Comment by capital-G** [comment]
Added tests and runs
```
youtube-dl$ python test/test_download.py TestDownload.test_Twitter_9
[twitter] 903676793860829184: Downloading webpage
[twitter:card] 903676793860829184: Downloading webpage
[twitter:card] 903676793860829184: Downloading m3u8 information
[twitter:card] 903676793860829184: Downloading mobile webpage
[twitter:card] 903676793860829184: Downloading main script
[twitter:card] 903676793860829184: Downloading mobile API data
[twitter:card] 903676793860829184: Downloading m3u8 information
[info] Writing video description metadata as JSON to: test_Twitter_9_903676793860829184.info.json
.
----------------------------------------------------------------------
Ran 1 test in 2.804s

OK
```

**Comment by capital-G** [comment]
Added duration to tests which returns a valid duration.
There are tests which do not return a duration - I have not took a deeper look at this as I just want to fix the scaling issue on duration, not extending the extractor with additional extraction logic for duration.

**Pull request merged**